coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/LinkedInOperatorBasedSqlCallTransformer.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino.transformers;
+
+import javax.annotation.Nonnull;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import com.linkedin.coral.common.transformers.OperatorBasedSqlCallTransformer;
+
+import static com.linkedin.coral.trino.rel2trino.CoralToTrinoSqlCallConverter.*;
+
+
+/**
+ * This is a subclass of OperatorBasedSqlCallTransformer which transforms a Coral operator into a Trino operator
+ * on SqlNode layer for the LinkedIn specific functions
+ */
+public class LinkedInOperatorBasedSqlCallTransformer extends OperatorBasedSqlCallTransformer {
+
+  public LinkedInOperatorBasedSqlCallTransformer(@Nonnull String linkedInFuncName, int numOperands,
+      @Nonnull String targetOpName) {
+    super(HIVE_FUNCTION_REGISTRY.lookup(linkedInFuncName).iterator().next().getSqlOperator(), numOperands,
+        targetOpName);
```

Comment:
Is there a risk of matching more than one entry in the `HIVE_FUNCTION_REGISTRY`? This assumes first match is returned. If we have multiple names with different number of parameters could this be a bug? cc: @ljfgem

Reply:
I think it shouldn't be an issue for current `StaticHiveFunctionRegistry`, one name only maps to one UDF (note that a UDF definition allows different operand numbers). But I think we can simplify HIVE_FUNCTION_REGISTRY to be a Map<String, Function> (with a seperate PR).